plugins/data-management/src/GridBookmarkWidget/components/DownloadBookmarks.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useState } from 'react'
+import { observer } from 'mobx-react'
+import { saveAs } from 'file-saver'
+
+import {
+  IconButton,
+  Button,
+  Dialog,
+  DialogTitle,
+  Select,
+  MenuItem,
+  makeStyles,
+} from '@material-ui/core'
+import CloseIcon from '@material-ui/icons/Close'
+import GetAppIcon from '@material-ui/icons/GetApp'
+
+import { getSession } from '@jbrowse/core/util'
+import { Region } from '@jbrowse/core/util/types'
+
+import { GridBookmarkModel } from '../model'
+
+function downloadBookmarkFile(bookmarkedRegions: Region[], fileFormat: string) {
+  const fileHeader =
+    fileFormat === 'TSV'
+      ? 'chrom\tstart\tend\tassembly_name\tcoord_range\n'
+      : ''
+
+  const fileContents = bookmarkedRegions
+    .map(b => {
+      if (fileFormat === 'BED') {
+        return `${b.refName}\t${b.start}\t${b.end}\n`
+      } else {
+        const locString = `${b.refName}:${b.start}..${b.end}`
+        return `${b.refName}\t${b.start}\t${b.end}\t${b.assemblyName}\t${locString}\n`
+      }
+    })
+    .reduce((a, b) => a + b, fileHeader)
+
+  const blob = new Blob([fileContents || ''], {
+    type:
```

Comment:
is the code sensitive to these MIME types?

Reply:
What do you mean by this?

Reply:
I just thought there was something specialized where the mime types needed to be special. I guess it probably isn't that consequential, but is there a source for how these were chosen?

Reply:
It just ends up telling the file downloader what the file type is when it's downloaded, works for me locally at least on MacOS.

Types were both found from Wikipedia:

- [BED](https://en.wikipedia.org/wiki/BED_(file_format))
- [TSV](https://en.wikipedia.org/wiki/Tab-separated_values)